indico/web/client/js/react/components/pictures/PictureArea.jsx

```diff
@@ -0,0 +1,240 @@
+                              />
+                            </span>
+                          }
+                          content={
+                            isCameraAllowed
+                              ? Translate.string('Take a picture with your camera')
+                              : Translate.string(
+                                  'You need to grant this site camera permissions before taking a picture'
+                                )
+                          }
+                        />
+                      </>
+                    ) : (
+                      <Grid columns={2} verticalAlign="middle" celled="internally" centered>
+                        <Grid.Column>
+                          <Button
+                            type="button"
+                            styleName="picture-selection-btn"
+                            icon={uploadButtonIcon}
+                            content={uploadButtonText}
+                            onClick={() => openUploadDialog()}
+                            disabled={dropzoneDisabled}
+                          />
+                        </Grid.Column>
+                        <Grid.Column>
+                          <Popup
+                            position="top center"
+                            trigger={
+                              <span>
+                                <Button
+                                  type="button"
+                                  styleName="picture-selection-btn"
+                                  icon={captureButtonIcon}
+                                  content={captureButtonText}
+                                  onClick={() => onOpenCameraDialog()}
+                                  disabled={!isCameraAllowed}
+                                />
+                              </span>
+                            }
+                            content={
+                              isCameraAllowed
+                                ? Translate.string('Take a picture with your camera')
+                                : Translate.string(
+                                    'You need to grant this site camera permissions before taking a picture'
+                                  )
+                            }
+                          />
+                        </Grid.Column>
+                      </Grid>
+                    )}
+                  </>
+                )}
+                <Divider horizontal />
+                <p>{descriptionText}</p>
+              </Grid.Column>
+            </Grid.Row>
+          </Grid>
+        )}
+        {(fileRejections.length > 0 || customFileRejections) && (
+          <Message negative>
+            <Message.Header>
+              <Translate>There were some problems with the picture</Translate>
+            </Message.Header>
+            {fileRejections.map(({file, errors}) => (
+              <Message.List key={file.size}>
```

Comment:
using the file size as the key is really strange. is there nothing better that's actually unique for each file?

Reply:
They're only passing the uploaded file and then the errors. I see examples where they use the file path as the key, is this acceptable?